tests/bdd/features/test_upgrade.py

```diff
@@ -0,0 +1,87 @@
+"""Upgrade feature tests."""
+
+import logging
+
+from common import helm, repo
+from common.environment import get_env
+from common.helm import ChartSource
+from common.kubectl_mayastor import kubectl_mayastor
+from kubernetes import client, config
+from pytest_bdd import given, scenario, then, when
+from retrying import retry
+
+logger = logging.getLogger(__name__)
+
+helm_client = helm.HelmReleaseClient()
+
+
+@scenario("upgrade.feature", "upgrade command is issued")
+def test_upgrade_command_is_issued():
+    """upgrade command is issued."""
+
+
+@given("an installed mayastor helm chart")
+def an_installed_mayastor_helm_chart():
+    """an installed mayastor helm chart."""
+    assert (
+        helm_client.install_mayastor(ChartSource.HOSTED, helm.latest_chart_so_far())
+        is not None
+    )
+
+
+@when("a kubectl mayastor upgrade command is issued")
+def a_kubectl_mayastor_upgrade_command_is_issued():
+    """a kubectl mayastor upgrade command is issued."""
+    assert kubectl_mayastor(["upgrade"]) is not None
+
+
+@then("the installed chart should be upgraded to the kubectl mayastor plugin's version")
+def the_installed_chart_should_be_upgraded_to_the_kubectl_mayastor_plugins_version():
+    """the installed chart should be upgraded to the kubectl mayastor plugin's version."""
+
+    upgrade_target_version = get_env("UPGRADE_TARGET_VERSION")
+    if upgrade_target_version is None:
+        upgrade_target_version = repo.run_script("scripts/python/generate-test-tag.sh")
+    upgrade_target_version = upgrade_target_version.lstrip("v")
+    logger.info(f"Value of upgrade_target_version={upgrade_target_version}")
+
+    @retry(
+        stop_max_attempt_number=15,
```

Comment:
Instead of checking for a maximum of 15 times at an interval of 1 min, I've now resorted to checking for a maximum of 450 times at an interval of 2 seconds.

On the other function...
Instead of checking for a maximum of 20 times at an interval of 1 min, I've now resorted to checking for a maximum of 600 times at an interval of 2 seconds.